### PR TITLE
Implement statuses controller

### PR DESCRIPTION
**Pull request by ihhyy**

### CODE REVIEW

File: InternshipBE/BL/Interfaces/ICandidateService.cs [hunk 1]
```diff
@@ -13,5 +14,7 @@ public interface ICandidateService
         Task<CandidateDTO> UpdateCandidateAsync(CandidateDTO updatedCandidate);
 
         Task<List<CandidateDTO>> GetCandidatesByInternshipIdAsync(int internshipId, int pageSize, int pageNumber);
+
+        Task<CandidateDTO> ChangeCandidateStatus(StatusType type, int Id);
```
Comment (romanstat) [inline InternshipBE/BL/Interfaces/ICandidateService.cs]: rename like another methods, swap parameters

File: InternshipBE/BL/Services/CandidateService.cs [hunk 2]
```diff
@@ -52,5 +53,15 @@ public async Task<List<CandidateDTO>> GetCandidatesByInternshipIdAsync(int inter
 
             return _mapper.Map<List<CandidateDTO>>(candidates);
         }
+
+        public async Task<CandidateDTO> ChangeCandidateStatus(StatusType type, int id)
+        {
+            var candidate = await _unitOfWork.Candidates.GetByIdAsync(id);
+            candidate.StatusType = type;
+            var mappedCandidate = _mapper.Map<Candidate>(candidate);
```
Comment (romanstat) [inline InternshipBE/BL/Services/CandidateService.cs]: candidate to candidate?

File: InternshipBE/WebApi/Controllers/StatusesController.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
+﻿using BL.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+using Shared.Enums;
+using System.Threading.Tasks;
+
+namespace WebApi.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class StatusesController : Controller
```
Comment (romanstat) [inline InternshipBE/WebApi/Controllers/StatusesController.cs]: you use candidateService as main, add method to CandidateController

File: InternshipBE/WebApi/Controllers/StatusesController.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    {
+        private readonly ICandidateService _candidateService;
+
+        public StatusesController(ICandidateService service)
+        {
+            _candidateService = service;
+        }
+
+        [HttpPut]
+        public async Task<IActionResult> UpdateCandidate(StatusType status, int id)
```
Comment (romanstat) [inline InternshipBE/WebApi/Controllers/StatusesController.cs]: same, swap parameters

**Review by romanstat — APPROVED** [review]

**Review by Nikityanov — APPROVED** [review]

**Review by vldmrklnk — APPROVED** [review]

**Pull request merged**